core/util/src/main/java/org/eclipse/rdf4j/common/iteration/ConvertingIteration.java

```diff
@@ -56,10 +58,18 @@ protected abstract T convert(S sourceObject)
 	 * @return <tt>true</tt> if the source type iteration contains more elements, <tt>false</tt> otherwise.
 	 * @throws X
 	 */
+	@Override
 	public boolean hasNext()
 		throws X
 	{
-		return iter.hasNext();
+		if (isClosed()) {
+			return false;
+		}
+		boolean result = iter.hasNext();
+		if (!result) {
+			close();
```

Comment:
I agree. At the time I thought the autoclose behavior was undocumented - which is why I accepted the criticism and abandoned the fix. But I later discovered that in fact it _was_ documented (and also implemented in several places, just not consistently). But I never got around to picking this up again.

So I'm good with this - it certainly makes everything more consistent. If we later find a good motivation for changing this we have a solid basis.